src/nss.c

```diff
@@ -35,6 +35,32 @@
 #include "util.h"
 #include "nss.h"
 
+#ifdef MDNS_MINIMAL
+#ifdef MDNS_ALLOW_FILE
+#  undef MDNS_ALLOW_FILE
+#endif
+#define MDNS_ALLOW_FILE NULL
+#endif
+
+static int avahi_get_af(int af, int unspec_ok) {
+    switch (af) {
+	/* intentionally fail build if both are defined */
+#ifndef NSS_IPV6_ONLY
+	case AF_INET:
+#endif
+#ifndef NSS_IPV4_ONLY
+	case AF_INET6:
+#endif
+	    return af;
+	case AF_UNSPEC:
```

Comment:
Missing previous functionality - if AF_UNSPEC and compiled with IPv4 only -> use AF for IPV4 etc.